src/storage-client/src/types/errors.rs

```diff
@@ -203,6 +203,32 @@ impl Display for UpsertValueError {
     }
 }
 
+/// A source contained a record with a NULL key, which we don't support.
+///
+/// NOTE: This does not contain additional (supposedly helpful) information, such as a partition ID
+/// or offset. We start out with keeping this error as generic as possible and, for example, a
+/// partition ID is a Kafka concept.
+#[derive(Ord, PartialOrd, Clone, Debug, Eq, PartialEq, Serialize, Deserialize, Hash)]
+pub struct UpsertNullKeyError {}
```

Comment:
what happens if we end up with a cardinality of more than `1` of these in the error collection?